airflow/operators/dagrun_operator.py

```diff
@@ -75,24 +63,21 @@ def __init__(
             self.execution_date = None
         else:
             raise TypeError(
-                'Expected str or datetime.datetime type '
-                'for execution_date. Got {}'.format(
-                    type(execution_date)))
+                "Expected str or datetime.datetime type for execution_date. "
+                "Got {}".format(type(execution_date))
+            )
 
     def execute(self, context):
```

Comment:
It's not a related change but since you are refactoring the operator I think adding a type hint does not hurt.
```suggestion
    def execute(self, context: Dict):
```